### PR TITLE
[Android] map view overhaul

### PR DESCRIPTION
**Pull request by felipecsl**
There are lots of ugly workarounds and random Android crashes that I'd like to get addressed but it's better if we start off from a clean slate.
* The fact that `AirMapView` extended `MapView` was both unnecessary and error-prone. I've converted it into a simple `FrameLayout` that adds the `MapView` as a child. This should help pinpoint any lifecycle issues since we can clearly separate between whats this library and what's Google Play Services maps.
* Reformatted the project to use the standard Airbnb java formatting (2 spaces). Added the settings file to the project.

This will likely introduce new crashes, but should be better in the long term so it's gonna be easier to maintain it in the future.

Best viewed with `?w=1` to ignore formatting changes.

### CODE REVIEW

**Comment by timxyz** [comment]
Aside from the code tidy-up, and the refactor, you have also removed code in AirMapView designed to fix an issue with a wrapped React context. I'm concerned this will cause regressions, as you now pass the React context straight to the Google MapView again. I wasn't involved with the original fix, so is there a reason you think the refactor will have fixed the issue?

**Comment by henrikra** [comment]
I would have put the beef of this PR to separate commit and style thing to other commit. Now it is very hard to see what was the actual thing in this PR :/

File: lib/android/src/main/java/com/airbnb/android/react/maps/MapsPackage.java [hunk 1]
```diff
@@ -13,41 +13,35 @@
...
+
+  @Override public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
+    AirMapCalloutManager calloutManager = new AirMapCalloutManager();
+    AirMapMarkerManager annotationManager = new AirMapMarkerManager();
+    AirMapPolylineManager polylineManager = new AirMapPolylineManager(reactContext);
+    AirMapPolygonManager polygonManager = new AirMapPolygonManager(reactContext);
+    AirMapCircleManager circleManager = new AirMapCircleManager(reactContext);
+
+    return Arrays.<ViewManager>asList(
+        calloutManager,
```
Comment (gpeal) [inline lib/android/src/main/java/com/airbnb/android/react/maps/MapsPackage.java]: Why not inline all of these?

**Comment by gpeal** [comment]
This is the latest crash:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Configuration android.content.res.Resources.getConfiguration()' on a null object reference
        at android.util.ResolutionOverride.<init>(ResolutionOverride.java:56)
        at android.view.SurfaceView.init(SurfaceView.java:207)
        at android.view.SurfaceView.<init>(SurfaceView.java:187)
        at com.google.maps.api.android.lib6.gmm6.vector.an.<init>(:com.google.android.gms.DynamiteModulesB:231)
        at com.google.maps.api.android.lib6.gmm6.vector.ai.<init>(:com.google.android.gms.DynamiteModulesB:163)
        at com.google.maps.api.android.lib6.gmm6.api.ao.<init>(:com.google.android.gms.DynamiteModulesB:389)
        at com.google.maps.api.android.lib6.gmm6.api.ao.a(:com.google.android.gms.DynamiteModulesB:349)
        at com.google.android.gms.maps.internal.eb.a(:com.google.android.gms.DynamiteModulesB:4048)
        at com.google.maps.api.android.lib6.impl.az.a(:com.google.android.gms.DynamiteModulesB:403)
        at com.google.maps.api.android.lib6.impl.az.a(:com.google.android.gms.DynamiteModulesB:308)
        at com.google.maps.api.android.lib6.impl.da.a(:com.google.android.gms.DynamiteModulesB:79)
        at com.google.android.gms.maps.internal.z.onTransact(:com.google.android.gms.DynamiteModulesB:66)
        at android.os.Binder.transact(Binder.java:387)
        at com.google.android.gms.maps.internal.IMapViewDelegate$zza$zza.onCreate(Unknown:-1)
        at com.google.android.gms.maps.MapView$zza.onCreate(Unknown:-1)
        at com.google.android.gms.dynamic.zza$3.zzb(Unknown:-1)
        at com.google.android.gms.dynamic.zza$1.zza(Unknown:-1)
        at com.google.android.gms.maps.MapView$zzb.zzJz(Unknown:-1)
        at com.google.android.gms.maps.MapView$zzb.zza(Unknown:-1)
        at com.google.android.gms.dynamic.zza.zza(Unknown:-1)
        at com.google.android.gms.dynamic.zza.onCreate(Unknown:-1)
        at com.google.android.gms.maps.MapView.onCreate(Unknown:-1)
        at com.airbnb.android.react.maps.AirMapView.<init>(AirMapView.java:126)
        at com.airbnb.android.react.maps.AirMapManager.createViewInstance(AirMapManager.java:62)
        at com.airbnb.android.react.maps.AirMapManager.createViewInstance(AirMapManager.java:29)
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:46)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:218)
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:148)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java:890)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:863)
        at com.facebook.react.uimanager.GuardedChoreographerFrameCallback.doFrame(GuardedChoreographerFrameCallback.java:32)
        at com.facebook.react.uimanager.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:131)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
        at android.view.Choreographer.doCallbacks(Choreographer.java:672)
        at android.view.Choreographer.doFrame(Choreographer.java:605)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:846)
        at android.os.Handler.handleCallback(Handler.java:739)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5441)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
```

**Comment by felipecsl** [comment]
You're right, let me break this down in 2 separate PRs!

**Pull request closed without merging**